recipes/triangle/meta.yaml

```diff
@@ -0,0 +1,67 @@
+{% set name = "triangle" %}
+{% set version = "20200424" %}
+{% set postfix = "wrapper" %}
+
+package:
+  name: {{ name|lower }}-{{ postfix|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: fc207641f8f39986f7d2bee1b91688a588cd235d2e67777422f94e61fece27e9
+
+build:
+  # noarch: python
+  number: 0
+  # If the installation is complex, or different between Unix and Windows, use separate bld.bat and build.sh files instead of this key.
+  # By default, the package will be built for the Python versions supported by conda-forge and for all major OSs.
+  # Add the line "skip: True  # [py<35]" (for example) to limit to Python 3.5 and newer, or "skip: True  # [not win]" to limit to Windows.
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  build:
+    # If your project compiles code (such as a C extension) then add the required compilers as separate entries here.
+    # Compilers are named 'c', 'cxx' and 'fortran'.
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - numpy
+    - cython >=0.24.0
+  run:
+    - numpy
+    - python
+
+test:
+  # Some packages might need a `test/commands` key to check CLI.
+  # List all the packages/modules that `run_test.py` imports.
+  imports:
+    - triangle
+
+about:
+  home: https://rufat.be/triangle
+  # Use the SPDX identifier, e.g: GPL-2.0-only instead of GNU General Public License version 2.0
+  # See https://spdx.org/licenses/
+  license: LGPL-3.0-only
```

Comment:
This is certainly not `LGPL`

Reply:
Hi,

whoops, I'm lost here: 

the python wrapper is definitely LGPL: https://github.com/drufat/triangle/blob/master/LICENSE

while the wrapped code is public domain http://www.cs.cmu.edu/~quake/triangle.html (also part of netlib) 

Does't that make the whole program LGPL?

Reply:
> while the wrapped code is public domain http://www.cs.cmu.edu/~quake/triangle.html (also part of netlib)

Not true. It has clauses like 

> These programs may be freely redistributed under the condition that the
copyright notices (including the copy of this notice in the code comments
and the copyright notice printed when the `-h' switch is selected) are
not removed, and no compensation is received.  Private, research, and
institutional use is free.

I am not an expert, but I think it means we can't have a python wrapper that is LGPL link with `triangle`.

cc @conda-forge/core.

Reply:
> I think it means we can't have a python wrapper that is LGPL link with triangle.

(usual disclaimer that I'm unqualified to comment) I think we can't have it in it's current form but we could have a separate `triangle` package which `py-triangle` dynamically links to? Conda-forge would then always ship the copyright notice and the "`copyright notice printed when the '-h' switch is selected`". Thoughts?

Reply:
Some more info that might help deciding what to do:

- The `triangle` package (of the c code) is already in conda https://anaconda.org/conda-forge/triangle and it also builds the library version of the code.

- On the github page of the wrapper https://github.com/drufat/triangle, the original c code has been moved to a different repo. (in PyPi, though, there's still an older version). However, in the `setup.py` the wrapper still compiles the c code, instead of using the compiled library. 

While it should be possible to follow the suggestion of @chisburr, I'm not sure this can be done with the present status of the wrapper. Maybe the author (cc @drufat) could help us with this?

Reply:
As it's already in conda-forge this is probably relatively simple to fix, just add a patch to the recipe to change the `setup.py` like so:
```diff
-        ['c/triangle.c', 'triangle/core.c'],
-        include_dirs=['c'],
+        ['triangle/core.c'],
+        libraries=["tri"],
```

Reply:
Cool, I wasn't aware of the possibility to add patches in the recipe. I'll try this way, thanks a lot.

Reply:
@chrisburr, it's still statically linking because only a static library is built.

Reply:
Ah I missed that...

Do you think it's acceptable to statically link and then include `triangle` as a (otherwise pointless) runtime dependency to comply with the license?